### PR TITLE
Implement CMSG_GROUP_SWAP_SUB_GROUP

### PR DESCRIPTION
**Pull request by DennisWG**
This Opcode allows to directly swap the sub groups of two players who are in the same raid. It's a lot easier to just show you what tis does, so here are some videos:

Before:
https://www.youtube.com/watch?v=xrh8U1idatg

After:
https://www.youtube.com/watch?v=mAv8HTVTKKk

### CODE REVIEW

**Comment by killerwife** [comment]
This wasn't implemented? Wow. Does this PR apply to both vanilla and wotlk?

**Comment by DennisWG** [comment]
I don't see a reason why the packet structure should've changed but I only have the TBC client installed so I can't check.

File: src/game/Groups/GroupHandler.cpp [hunk 1]
```diff
@@ -886,3 +886,44 @@ void WorldSession::HandleOptOutOfLootOpcode(WorldPacket& recv_data)
...
+        else
+        {
+            if (ObjectGuid guid = sObjectMgr.GetPlayerGuidByName(playerName))
+                return guid;
+            else
+                return ObjectGuid();
+        }
+    };
+
+    ObjectGuid guid1 = getGuid(playerName1);
```
Comment (Warlockbugs) [inline src/game/Groups/GroupHandler.cpp]: Exiting the function as soon as either name query fails may be useful. One full name search can be avoided if first name is unknown.
```
if (!guid)
    return;
```

**Comment by cala** [comment]
The feature is required in Classic and currently missing. I can't tell if the packet structure or even OPCode is the same thought.
But I don't see what changes occurred between Classic and TBC that would have require it to be different.

**Comment by Warlockbugs** [comment]
Checked classic, packet structure is the same, so this patch can be applied without any alterations.

**Pull request merged**